pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -337,16 +344,18 @@ private void processResult(Record srcRecord,
         });
     }
 
-    private void sendOutputMessage(Record srcRecord, Object output) {
+    private void sendOutputMessage(Record srcRecord, Object output) throws Exception {
         if (!(this.sink instanceof PulsarSink)) {
             Thread.currentThread().setContextClassLoader(functionClassLoader);
         }
         try {
             this.sink.write(new SinkRecord<>(srcRecord, output));
         } catch (Exception e) {
+        	if (stats != null) {
```

Comment:
Added null check to make it consistent with the `readInput()` method exception handling